bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -454,22 +454,27 @@ where
 	}
 }
 
-pub struct GlobalConsensusEthereumConvertsFor<AccountId>(PhantomData<AccountId>);
-impl<AccountId> ConvertLocation<AccountId> for GlobalConsensusEthereumConvertsFor<AccountId>
+pub struct EthereumLocationsConverterFor<AccountId>(PhantomData<AccountId>);
+impl<AccountId> ConvertLocation<AccountId> for EthereumLocationsConverterFor<AccountId>
 where
 	AccountId: From<[u8; 32]> + Clone,
 {
 	fn convert_location(location: &Location) -> Option<AccountId> {
 		match location.unpack() {
-			(_, [GlobalConsensus(Ethereum { chain_id })]) =>
+			(2, [GlobalConsensus(Ethereum { chain_id })]) =>
```

Comment:
Why be explicit on the parents here instead? Wouldnt this limit its usage to being relative to parachains?

It would be parents 1 if relative to the relaychain. It would also be parents 1 if relative to another Ethereum chain. This wont break any existing use cases though so maybe its fine.

Reply:
Sorry, I forgot to mention this as well in the description (edited now).

I prefer tightening the match to explicit `parents: 2` since this converter is meant to be used by parachains only. I can't think of a valid (long-term) use case for the Relay chain to need to derive Ethereum locations.

> It would also be parents 1 if relative to another Ethereum chain.

You mean if you use this converter in an Ethereum SC? I don't think there is such a real case, but if there will ever be I suggest adding a new entry to the match there or even using a separate converter since they would be different in output anyway (32bytes vs 20bytes).

Reply:
> It would be parents 1 if relative to the relaychain

@alistair-singh I think what you mentioned is another [convert](https://github.com/paritytech/polkadot-sdk/blob/5a142856520fd450b6be8361a50e1f4d385e8a6c/bridges/snowbridge/primitives/core/src/location.rs#L58) to convert a token location of PNA to a stable ID.

It's nothing to do with the above change, it's for ENA and should be fine to set parents to 2.